### PR TITLE
WIP - initial commit hono-kura adapter folder structure

### PR DESCRIPTION
**Pull request by oafd**
#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [ ] [DCO](https://developercertificate.org) signed
- [ ] Chart Version bumped (if the pr is an update to an existing chart)
- [ ] Variables are documented in the README.md
- [ ] Title of the PR starts with chart name (e.g. `[fluentd-elasticsearch]`)

### CODE REVIEW

**Comment by rpahli** [comment]
What is this PR for... I  think this is a copy for the original chart?  Do we really want to maintain this chart in our repo?

**Comment by axdotl** [comment]
@rpahli The original chart has some drawbacks:
* It's not flexible enough. As example, it is not possible to en-/disable single adapaters
* It's not available in a public repository, only as part of the archive available here: https://www.eclipse.org/hono/downloads/
* It's not following best practices (e.g. standard labels are missing)

File: charts/hono-adapter-kura/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+appVersion: "1.0-M5"
+description: A Helm chart for the Hono-Kura adapter
+name: hono-adapter-kura
+version: 0.1.0
```
Comment (axdotl) [inline charts/hono-adapter-kura/Chart.yaml]: Icon is missing, you could use: https://raw.githubusercontent.com/eclipse/hono/master/site/documentation/static/images/logo_300px276px.png

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    matchLabels:
+     app.kubernetes.io/name: {{ include "hono-adapter-kura.name" . }}
+     app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "hono-adapter-kura.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- if eq .Values.jaeger.enabled true}}
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: I suggest to omit jaeger support for now

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      initContainers:
+        - name: wait-for-jaeger-agent
+          image: busybox:1.28
+          command: ['sh', '-c', 'until nslookup {{ .Values.jaeger.agentHost }}; do echo waiting for {{ .Values.jaeger.agentHost }}; sleep 2; done;']
+      {{- end}}
+      containers:
+      - image: {{ .Values.honoContainerRegistry }}/eclipse/hono-adapter-kura:{{ .Values.honoImageTag }}
+        imagePullPolicy: IfNotPresent
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: has to be configurable

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+          value: classpath:logback-spring.xml
+        {{- if eq .Values.jaeger.enabled true}}
+        - name: JAEGER_SERVICE_NAME
+          value: hono-adapter-kura
+        {{- range $key, $value := .Values.jaeger.env }}
+        - name: {{ $key }}
+          value: {{ $value | quote }}
+        {{- end }}
+        {{- end}}
+        - name: _JAVA_OPTIONS
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: Support for arbitrary env variables is recommended

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+          value: "{{ .Values.defaultJavaOptions }}"
+        - name: KUBERNETES_NAMESPACE
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: Unnecessary

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+          readOnly: true
+        resources:
+          limits:
+            memory: "256Mi"
+        readinessProbe:
+          httpGet:
+            path: /readiness
+            port: {{ .Values.healthCheckPort }}
+            scheme: HTTP
+          initialDelaySeconds: 10
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: Probe configuration has to be configurable

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        livenessProbe:
+          httpGet:
+            path: /liveness
+            port: {{ .Values.healthCheckPort }}
+            scheme: HTTP
+          initialDelaySeconds: 180
+      volumes:
+      - name: conf
+        secret:
+          secretName: hono-adapter-kura-conf
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: This secret does not exits as long as you not provide it.

File: charts/hono-adapter-kura/templates/service.yaml [hunk 8]
```diff
@@ -0,0 +1,28 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ include "hono-adapter-kura.fullname" . }}
+  labels:
+{{ include "hono-adapter-kura.labels" . | indent 4 }}
+
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/service.yaml]: remove empty line

File: charts/hono-adapter-kura/templates/service.yaml [hunk 9]
```diff
@@ -0,0 +1,28 @@
...
+spec:
+  ports:
+  - name: mqtt
+    port: 1883
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/service.yaml]: ports should be configurable (true for all listed ports)

File: charts/hono-adapter-kura/templates/service.yaml [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  - name: secure-mqtt
+    port: 8883
+    protocol: TCP
+    targetPort: secure-mqtt
+    nodePort: 30884
+  selector:
+    app.kubernetes.io/name: {{ include "hono-adapter-kura.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    group: {{ .Values.project.groupId }}
+  {{- if and ( eq .Values.useLoadBalancer true ) ( ne .Values.platform "openshift" )}}
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/service.yaml]: No openshift support

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 11]
```diff
@@ -0,0 +1,84 @@
...
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
+      {{- if eq .Values.jaeger.enabled true}}
+      initContainers:
+        - name: wait-for-jaeger-agent
+          image: busybox:1.28
+          command: ['sh', '-c', 'until nslookup {{ .Values.jaeger.agentHost }}; do echo waiting for {{ .Values.jaeger.agentHost }}; sleep 2; done;']
+      {{- end}}
+      containers:
+      - image: {{ .Values.honoContainerRegistry }}/eclipse/hono-adapter-kura:{{ .Values.honoImageTag }}
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: The referenced property keys are invalid

File: charts/hono-adapter-kura/templates/deployment.yaml [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+  selector:
+    matchLabels:
+     app.kubernetes.io/name: {{ include "hono-adapter-kura.name" . }}
+     app.kubernetes.io/instance: {{ .Release.Name }}
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/name: {{ include "hono-adapter-kura.name" . }}
+        app.kubernetes.io/instance: {{ .Release.Name }}
+    spec:
```
Comment (axdotl) [inline charts/hono-adapter-kura/templates/deployment.yaml]: `imagePullSecrets` has to be added

**Comment by monotek** [comment]
> @rpahli The original chart has some drawbacks:
> 
> * It's not flexible enough. As example, it is not possible to en-/disable single adapaters/ components
> * It's not available in a public registry, only as part of the archive available here: https://www.eclipse.org/hono/downloads/
> * It's not following best practices (e.g. standard labels are missing)

did we tried to contribute first?

File: charts/hono-adapter-kura/Chart.yaml (hunk 1 above)
Comment (monotek) [inline charts/hono-adapter-kura/Chart.yaml]: "home" is missing

File: charts/hono-adapter-kura/Chart.yaml [hunk 13]
```diff
@@ -0,0 +1,10 @@
...
+sources:
+- https://github.com/eclipse/hono/tree/master/adapters/kura
+maintainers:
+- name: oafd
```
Comment (monotek) [inline charts/hono-adapter-kura/Chart.yaml]: imho we should have a second maintainer in the chart

File: charts/hono-adapter-kura/Chart.yaml (hunk 1 above)
Comment (oafd) [reply charts/hono-adapter-kura/Chart.yaml]: ok

**Comment by monotek** [comment]
@oafd 
whats the progress here?

**Comment by oafd** [comment]
> @oafd
> whats the progress here?
@monotek , no progress so far, it was postponed! the test will happen locally. This testing will be covered with @rpahli's kura-hono-pubsub overall test

**Pull request closed without merging**